addons/product/static/src/js/product_configurator/product_configurator_dialog/product_configurator_dialog.js

```diff
@@ -0,0 +1,397 @@
+        productUOMId: { type: Number, optional: true },
+        companyId: { type: Number, optional: true },
+        edit: { type: Boolean, optional: true },
+        save: Function,
+        discard: Function,
+        close: Function, // This is the close from the env of the Dialog Component
+        options: {
+            type: Object,
+            shape: {
+                no_open : { type: Boolean, optional: true },
+                showCustom: { type: Boolean, optional: true },
+                showQty: { type: Boolean, optional: true },
+                qtyFieldName: {
+                    type: String,
+                    optional: true,
+                }, // Some models have different fields names for qty
+                productUomIdFieldName: {
+                    type: String,
+                    optional: true,
+                }, // Some models have different fields names for product_uom
+            }
+        },
+    };
+    static defaultProps = {
+        edit: false,
+    }
+
+    setup() {
+        this.title = _t("Configure");
+        this.rpc = useService("rpc");
+        this.state = useState({
+            products: [],
+        });
+
+        useSubEnv({
+            mainProductTmplId: this.props.productTemplateId,
+            setQuantity: this._setQuantity.bind(this),
+            updateProductTemplateSelectedPTAV: this._updateProductTemplateSelectedPTAV.bind(this),
+            updatePTAVCustomValue: this._updatePTAVCustomValue.bind(this),
+            isPossibleCombination: this._isPossibleCombination,
+        });
+
+        onWillStart(async () => {
+            await this._setProducts();
+            for (const customValue of this.props.customAttributeValues) {
+                this._updatePTAVCustomValue(
+                    this.env.mainProductTmplId,
+                    customValue.ptavId,
+                    customValue.value
+                );
+            }
+            await this._onWillStartCustomCheck();
+            this._checkExclusions(this.state.products[0]);
+        });
+    }
+
+    async _onWillStartCustomCheck(){
+        if(this.props.options?.showCustom){
+            return
+        }
+        for (let product of this.state.products){
+            for (let attribute_line of product.attribute_lines){
+                const firstNotCustom = attribute_line.attribute_values.filter(
+                    val => !val.is_custom);
+                if(firstNotCustom.length > 0){
+                    await this._updateProductTemplateSelectedPTAV(product.product_tmpl_id, attribute_line.id, firstNotCustom[0].id);
+                }
+            }
+        }
+    }
+
+    //--------------------------------------------------------------------------
+    // Data Exchanges
+    //--------------------------------------------------------------------------
+
+    async _loadData(onlyMainProduct) {
+        return this.rpc('/product_configurator/get_values', this._loadDataVals(onlyMainProduct));
```

Comment:
I'd rather keep the needed parameters inside the call and use a hook to add additional parameters. I don't want modules that extends this method to be able to remove the basic parameters